app/scripts/modules/core/src/pipeline/config/stages/awsCodeBuild/AwsCodeBuildStageForm.tsx

```diff
@@ -114,6 +121,22 @@ export function AwsCodeBuildStageForm(props: IAwsCodeBuildStageFormProps & IForm
         name="sourceVersion"
         input={(inputProps: IFormInputProps) => <TextInput {...inputProps} />}
       />
+      <FormikFormField
+        fastField={false}
+        label="Buildspec"
+        name="buildspec"
+        input={(inputProps: IFormInputProps) => (
+          <YamlEditor {...inputProps} value={get(stage, 'buildspec')} onChange={onYamlChange} />
+        )}
+      />
+      <h4>Environment Configuration</h4>
+      <FormikFormField
+        fastField={false}
+        label="Image"
+        name="image"
+        input={(inputProps: IFormInputProps) => <TextInput {...inputProps} />}
+      />
```

Comment:
As a todo, it'd be neat to use the Docker image selector, so you can get a dropdown of images to select from

Reply:
I will take a look at the Docker image selector to see if it could be used here

Reply:
I looked at the `DockerImageAndTagSelector`. This is used for selecting an image but not matching against the incoming docker images. The image selected will be fixed. For example, if I configured 2 triggers for 2 different images, this selector could only select one of them. No matter which image triggers the pipeline, CodeBuild will always build using the image selected, which is definitely not what we want. For this `TextInput` approach, user has the ability by specifying `${trigger['repository']}/${trigger['tag']}` in this field.